packages/runtime/container-runtime/src/summarizerTypes.ts

```diff
@@ -34,6 +33,12 @@ export interface IProvideSummarizer {
     readonly ISummarizer: ISummarizer;
 }
 
+/* Similar to AbortSignal, but using promise instead of events */
+export interface ICancellable {
+    readonly cancelled: boolean;
+    readonly waitCancelled: Promise<void>;
```

Comment:
can you add comments? i don't understand this promise? will it always resolve? should it be waitFinished or something?

Reply:
I'll add comments. It is resolved when this object is cancelled.